core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -434,6 +434,10 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
               records.foreach(producer.send)
               maybeFlushAndCommitOffsets()
             }
+            if (!mirrorMakerConsumer.hasData && !shuttingDown) {
```

Comment:
a send failure closes the producer, and not the consumer, meaning mirrorMakerConsumer.hasData will return true and this wont be called. its possible for there to be both a send failure and consumer death at the same time, in which case which one kills the mm is semantics :-). will update.